lua/entities/gmod_wire_expression2/core/sound.lua

```diff
@@ -66,26 +67,73 @@ local function soundStop(self, index, fade)
 	timer.Remove( "E2_sound_stop_" .. self.entity:EntIndex() .. "_" .. index )
 end
 
+local ClientSideSound = {}
+ClientSideSound.mt = {__index = ClientSideSound}
+
+ClientSideSound.SendIndex = function(self)
+	umsg.String(self.e2:EntIndex() .. "_" .. self.index)
```

Comment:
There's really no reason to use umsg over net. You should switch to net messages.